src/web/lib/components/ThemeCustomBackgroundPicker/index.js

```diff
@@ -298,6 +298,8 @@ class ThemeCustomBackgroundSelector extends React.Component {
 
               <ImportButton label={errors ? "Retry" : "Replace image"} />
 
+              {this.props.displayRemoveImageModal && <div className="modal-wrapper--clear-image"><ClearImageModal {...this.props} /></div>}
```

Comment:
Rather than blindly forwarding `this.props` and duplicating the names "clearImageModal", `clearCustomBackground`, `setDisplayRemoveImageModal` in two different files, would it be possible to keep the logic in one place?

For example, the modal file:

```
<button onClick={this.props.confirm}>...
<button onClick={this.props.cancel}>...
```

and here:

```
<ClearImageModal
  confirm={() => this.props.clearCustomBackground() }
  cancel={() => this.props.setDisplayRemoveImageModal({ display: true }) }
>
```

Reply:
hey @Rob--W , I definitely like the idea of keeping this in one 1 file I just thought it was better to set the localStorage key `clearImageModal` in the modal itself. If I change it, I would have something like this in the ThemeCustomBackgroundPicker file:

```javascript
  <ClearImageModal
     confirm={this.confirm}
     cancel={this.onCloseModal}
  />

...

  confirm = () => {
    this.props.clearCustomBackground();
    this.onCloseModal();
  }

  onCloseModal = () => {
    // the following needs to happen on "okay" or "cancel"
    this.props.setDisplayRemoveImageModal({ display: false }); 
    localStorage.setItem("clearImageModal", true);
  }

  handleClearBackground = () => {
    if (localStorage.getItem("clearImageModal")) {
      this.props.clearCustomBackground();
    } else {
      this.props.setDisplayRemoveImageModal({ display: true });
    }
  };
```

if you agree that setting the `clearImageModal` localStorage value here is fine (instead of in the `ClearImageModal` component), then I'll go ahead and move this as shown.

Reply:
Otherwise, I'll keep it as I currently have it. but wanted to know your thoughts before I changed/merged anything :)
thanks!

Reply:
The snippet that you described looks good to me.